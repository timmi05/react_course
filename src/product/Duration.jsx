import React from 'react';
import {DigitalInfo} from '../component/DigitalInfo';

export const Duration = props => {
    const {runtime} = props;
    if (runtime === 0) {
        return (
            <a/>
        )
    }
    return (
        <DigitalInfo
            value={runtime}
            meaning=' min'
        />
    )
};