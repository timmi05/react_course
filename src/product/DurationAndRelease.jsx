import React from 'react';
import DigitalInfo from '../component/DigitalInfo';

const DurationAndRelease = (props) =>{
    const { runtime, release_date } = props;
    return (
         <div className="duration-and-release">
            <DigitalInfo
            value = { runtime }
            meaning = 'year'
            />
            <DigitalInfo
            value = {runtime}
            meaning = 'min'
            />
         </div>
    );
}

export default DurationAndRelease