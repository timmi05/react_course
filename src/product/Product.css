.product{
    display: flex;
    flex-direction: row;
    margin-bottom: 24px;
    margin-top: 24px;
}

.product-cover{
    height: 450px;
    width: 315px;
}