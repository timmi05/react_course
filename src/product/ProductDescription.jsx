import React from 'react';

export const ProductDescription = (props) => {
   const { overview } = props;
   return (
   <span className='product-description'>{ overview }</span>
    );
};

export default ProductDescription