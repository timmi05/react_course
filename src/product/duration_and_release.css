.duration-and-release{
    color: #f65261;
    font-size: 26px;
    margin-bottom: 25px;
}

.duration-and-release-release{
    margin-right: 30px;
}