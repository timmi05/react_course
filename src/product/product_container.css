.product-container{
    display: flex;
    flex-direction: row;
}

.product-container-cover{
    height: 400px;
    width: 250px;
}