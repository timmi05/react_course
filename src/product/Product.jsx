import React from 'react';
import {ProductInfo} from './ProductInfo';
import './Product.css';

export const Product = props =>
    <div className='product'>
        <img className='product-cover' src={props.movie.poster_path} alt={props.movie.title}/>
        <ProductInfo
            {...props}
        />
    </div>;
