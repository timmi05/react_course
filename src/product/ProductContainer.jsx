import React from 'react';
import ProductInfo from './ProductInfo';
import './product_container.css';

const ProductContainer = (props) => {
        const { poster_path, title } = props;
        return (
             <div className='product-container'>
             <img className='product-container-cover' src={ poster_path } alt={ title }/>
             <ProductInfo {...props}/>
            </div>
        );
};
export default ProductContainer