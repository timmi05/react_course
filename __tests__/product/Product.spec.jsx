import React from 'react';
import {shallow} from 'enzyme';
import {Product} from '../../src/product/Product';

describe('Product rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<Product
            movie={[]}
        />);
        expect(component).toMatchSnapshot();
    });
});
