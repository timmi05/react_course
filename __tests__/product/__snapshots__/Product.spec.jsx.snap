// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product rendering renders correctly 1`] = `
<div
  className="product"
>
  <img
    className="product-cover"
  />
  <ProductInfo
    movie={Array []}
  />
</div>
`;
